addon/utils/proxy-unproxy-objects.js

```diff
@@ -0,0 +1,46 @@
+import { isNone } from '@ember/utils';
+/**
+ * This utility is ussed to create a wrapper object around the object passed to the proxy function.
+ * Currently on drag action, the `draggable-object` mutates the `content` object and appends a new
+ * property `isDraggingObject` onto the parent object.
+ * This behavior throws an exception with M3 models because it not a previously whitelisted property. The issue
+ * is being tracked (ticket: TODO: Add Github issue ticket here).
+ *
+ * This utility is used to create a proxy object to wrap the original `content` M3 objet to 
+ * avoud any exceptions from being thrown when dragging is initiated on the object.
+ * 
+ * Note: Added further implementation to `unproxyObject` that can be used to unproxy the previously
+ * proxies object thereby returning the original M3 object.
+ *
+ */
+
+/**
+ * @access public
+ * Returns the passed object wrapped within new object.
+ * Used to proxy draggable objects.
+ * @param objectToProxy Object to proxy.
+ * @returns {Object} Proxy object.
+ */
+export function proxyObject(objectToProxy) {
+  if (!isNone(objectToProxy)) {
+    return {
```

Comment:
since we aren't returning a true proxy but rather a wrapper (e.g. there is no property forwarding) we should call this either a `wrapper` or a `manager`

Reply:
+1